reframe/core/pipeline.py

```diff
@@ -925,7 +927,9 @@ def _process_hook_registry(cls):
 
         _pipeline_hooks = {}
         for stage, hooks in cls.pipeline_hooks().items():
-            stage_name = '_'.join(stage.split('_')[1:])
+            # Pop the stage pre_/post_ prefix
+            stage_name = re.match(r'p\w{2,3}_(\w+)', stage)[1]
```

Comment:
I'd avoid the fuss of regexes here. The following should cover also the cases where stage names have underscores.

```suggestion
            stage_name = stage.split('_', maxsplit=1)[1]
```